app/models/manageiq/providers/kubernetes/container_manager/metrics_capture/hawkular_client_mixin.rb

```diff
@@ -1,19 +1,26 @@
 module ManageIQ::Providers::Kubernetes::ContainerManager::MetricsCapture::HawkularClientMixin
   def hawkular_client
     require 'hawkular/hawkular_client'
-    @client ||= Hawkular::Metrics::Client.new(
-      hawkular_entrypoint, hawkular_credentials, hawkular_options)
+    @client ||= Hawkular::Metrics::Client.new(hawkular_entrypoint, hawkular_credentials, hawkular_options)
   end
 
   def hawkular_entrypoint
     hawkular_endpoint = @ext_management_system.connection_configurations.hawkular.try(:endpoint)
     hawkular_endpoint_empty = hawkular_endpoint.try(:hostname).blank?
+    if hawkular_endpoint_empty
+      # if hawkular endpoint left empty learn the hostname from the hawkular route
+      hawkular_route = ems.connect.get_routes.select do |elem|
```

Comment:
AFAIK you better use `find_by_name` instead of select, its more readable and also I suspect runs faster.